rules/library.bzl

```diff
@@ -492,26 +621,33 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
                 **kwargs
             )
 
-        framework_vfs_overlay_name = name + "_vfs"
-        framework_vfs_overlay(
-            name = framework_vfs_overlay_name,
-            framework_name = namespace,
-            modulemap = module_map,
-            private_hdrs = objc_private_hdrs,
-            hdrs = objc_hdrs,
-            tags = _MANUAL,
-        )
-        private_deps.append(framework_vfs_overlay_name)
-        additional_objc_copts += [
-            "-ivfsoverlay$(execpath :{})".format(framework_vfs_overlay_name),
-            "-F{}".format(VFS_OVERLAY_FRAMEWORK_SEARCH_PATH),
-        ]
-        additional_swift_copts += [
-            "-Xcc",
-            "-ivfsoverlay$(execpath :{})".format(framework_vfs_overlay_name),
-            "-Xcc",
-            "-F{}".format(VFS_OVERLAY_FRAMEWORK_SEARCH_PATH),
-        ]
+    framework_vfs_overlay(
+        name = framework_vfs_overlay_name_swift,
+        framework_name = module_name,
+        modulemap = module_map,
+        has_swift = len(swift_sources) > 0,
+        private_hdrs = objc_private_hdrs,
+        hdrs = objc_hdrs,
+        tags = _MANUAL,
+        deps = deps + private_deps + lib_names + import_vfsoverlays,
+    )
+    private_deps.append(framework_vfs_overlay_name_swift)
+
+    additional_objc_copts += [
+        "-ivfsoverlay$(execpath :{})".format(framework_vfs_overlay_name),
+        "-F{}".format(VFS_OVERLAY_FRAMEWORK_SEARCH_PATH),
+    ]
+    additional_swift_copts += [
+        "-Xfrontend",
+        "-vfsoverlay$(execpath :{})".format(framework_vfs_overlay_name_swift),
+        "-Xfrontend",
+        "-F{}".format(VFS_OVERLAY_FRAMEWORK_SEARCH_PATH),
```

Comment:
🎉  I need to do a before after experiment on building some modules to see how the compilation params differ based on these

Reply:
The compilation parameters are certainly different here - it now passes the VFS directly to the frontend